project_2/defore_dir/renaming_file_name.py

```diff
@@ -0,0 +1,48 @@
+import os
+from pathlib import Path
+import argparse
+from collections import namedtuple
+from logger_info import log_all
+
+
+def re_file(dir_name):
+    # dir_name = input("Укажите путь до директории:")
+    if os.path.isdir(dir_name):  # само определяет как выставить слеши в пути
+        obj_list = []
+
+        p = Path(dir_name)  # инфо из текущей директории
+        for obj in p.iterdir():  # перебор из этой директории
+            # print(obj)
+            obj = str(obj).lower().split('\\')  # приведение к нижнему регистру т.к расширение может быть заглавным
+            # print(obj)
+            if '.' in obj[-1]:  # если в последнем элементе из директории есть точка значит есть и расширение
+                temp_obj = obj[-1].split('.')  # разделяем элемент по этой точке
+                end_ex = temp_obj[-1]  # Будет расширением
+                name_file = temp_obj[0]  # Будет именем файла
+                obj.pop(-1)  # Будет родительской директорией для файла или папки
+                temp = ('\\'.join(obj), name_file, end_ex)  # собираем кортеж данных,ведь изменения в пути не вносим
+                obj_list.append(temp)  # добавляем в список результат из текущей директории
+            else:
+                name_file = 'false'  # Вместо флагов если нет расширений
+                end_ex = 'false'
+                obj.pop(-1)
+                temp = ('\\'.join(obj), name_file, end_ex)
+                obj_list.append(temp)
+        copy_name_list = []  # Будет лист экземпляров
+        print(obj_list)
+        for item in obj_list:
+            a, b, c = item  # Распоковываем  каждый кортеж на значения
+            DIR_LIST = namedtuple('DIR_LIST', ['dir', 'file', 'exeption'])  # Делаем класс со свойствами
```

Comment:
Наверное `extension`
```suggestion
            DIR_LIST = namedtuple('DIR_LIST', ['dir', 'file', 'extension'])  # Делаем класс со свойствами
```